### PR TITLE
Specify package Metadata using a pyproject.toml file

### LINKED ISSUE #49: Bring `pyproject.toml` Back!
@vighneshiyer pointed out (some time ago now) that `Poetry` now supports the feature we really wanted it to, when we ditched it: editable dev-mode installs. 

I think it's worth bringing back, when we have the chance.

### LINKED ISSUE #218: BUG: sky130 pdk pypi Install
When I try to install the sky130 pdk I receive the following error due to it not being able to find the readme.  Here I have "sky130-hdl21 == 5.0.0" as a dependency in the pyproject.toml of my "bandgaps" project and am trying to do an editable install of bandgaps.

I would propose that we convert the sky130-hdl21 pdk package to use a pyproject.toml to define the metadata and build the project using [flit](https://flit.pypa.io/en/stable/).  

This would have a simpler definition than the currernt setup.py and handles problems where the package is a zip rather than an actual path.

image

### PR DESCRIPTION
**Pull request by curtisma**
Closes #218 and closes #49 

## Updates
- I updated all the packages to specify their metadata in a [pyproject.toml](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#writing-your-pyproject-toml) which replaced all setup.py files.

- Updated the package build backend to [flit ](https://flit.pypa.io/en/stable/)instead of using [build](https://build.pypa.io/en/stable/) as the package build back-end.

- Updated the sitepdks package name from "sitepdks" to "hdl21-sitepdks"

- The Pytest configuration was merged into the [pyproject.toml from the pytest.ini](https://docs.pytest.org/en/7.3.x/reference/customize.html#pyproject-toml).

## Questions & Concerns

- We should add the rest of the contributors to the authors list.
  @ThomasPluck @kcaisley @uduse @growly @vighneshiyer @uduse @aviralpandey
  Do you guys mind if I add your name and e-mail to the authors list?
  If so please reply with the name and e-mail you would like included.  We can also just include your name if you'd prefer not to include your e-mail.  I added some of your names.
  

- I dropped the twine dev dependency.  Flit also has a ["flit publish" feature](https://flit.pypa.io/en/stable/cmdline.html#flit-publish) that can do the same thing as twine.

- I couldn't share the version amongst all the packages to tie them to the vlsir version number.  Shouldn't these versions be updated independently?  We shouldn't publish a new version of a package unless something has changed.  We can just provide a major version range for the related packages.  I went ahead and updated the version range to peg them to a major revision of the related projects, e.g. vlsir and hdl21.  They're also in different states of maturity.  I would think that some of the less mature packages (e.g. the asap7 pdk)  should still be in a v0.x.x version number to reflect its immaturity.  It would be good to follow the [semantic versioning recommendation](https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase) to keep a package at "v0.y.z" until the api is stable.

- Should we upgrade from black to [ruff ](https://docs.astral.sh/ruff/) in the dev depencies?  It's faster and has more features.

## Testing

Tested both the pdk and main pyproject.toml by doing an install and editable install of each:

image

image

image

### CODE REVIEW

**Comment by curtisma** [comment]
I still need to add a LICENSE file to the root of each package.  That will make sure it is included properly in the package and would be more explicit that all the packages use bsd3.

**Comment by curtisma** [comment]
> I still need to add a LICENSE file to the root of each package.  That will make sure it is included properly in the package and would be more explicit that all the packages fall under a bsd3 license.

Done

**Comment by daquintero** [comment]
Just thought to write, thanks for this effort! Installation works for me from a poetry-managed environment with your commit with the new pyproject.toml.

I've just run this successfully based on this branch within pydantic v2 in Ubuntu py 3.10. Working on some examples so if I run into any issues will raise/fix as part of the pydantic v2 new release. Look forward to trying it out in the wild directly from pypi!
```
import hdl21 as h
import sky130_hdl21

@h.module
class SkyInv:
    """ An inverter, demonstrating using PDK modules """

    # Create some IO
    i, o, VDD, VSS = h.Ports(4)

    p = sky130_hdl21.Sky130MosParams(w=1,l=1)

    # And create some transistors!
    ps = sky130_hdl21.primitives.PMOS_1p8V_STD(p)(d=o, g=i, s=VDD, b=VDD)
    ns = sky130_hdl21.primitives.NMOS_1p8V_STD(p)(d=o, g=i, s=VSS, b=VSS)
```

**Comment by dan-fritchman** [comment]
CI fails, although (I think) only on a code-coverage upload step. (There's a lotta tests.)  
Which (again I think) is because the source branch is on a fork?  
We should test that, and probably make an issue for it.

File: SampleSitePdks/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+[project]
+name = "hdl21-sitepdks"
```
Comment (dan-fritchman) [inline SampleSitePdks/pyproject.toml:2]: Keep the name `sitepdks` plz

**Comment by dan-fritchman** [comment]
OK never mind that codecov/ CI comment.  
Seems it was a transient thing that resolved itself.  
Both the re-run CI here and those on #221 (same code) both pass.

File: SampleSitePdks/pyproject.toml (hunk 1 above)
Comment (curtisma) [reply SampleSitePdks/pyproject.toml]: ok, updated.  Please approve the CI run.

**Pull request merged**